### ISSUE TITLE
fx>= has broken type

### ISSUE THREAD

**Issue by endobson** [racket/typed-racket#115]
This typechecks:
```
#lang typed/racket
 
(require racket/flonum)

(ann
  (let ([x : Flonum 5.0])
    (if 
      (fl>= x (ann -4.0 Nonpositive-Flonum))
      x
      -3.0))
  Nonpositive-Flonum)
```

**Comment by shekari** [comment]
This is also true for extfl>=.

**Comment by takikawa** [comment]
I think this was fixed already, possibly in commit 7e32ec93eb3b0b32a8a88097630a1527740b3a0e

**Comment by rfindler** [comment]
I don't see any test cases.....
